dspace-api/src/main/java/org/dspace/submit/step/UploadStep.java

```diff
@@ -614,7 +614,7 @@ public int processUploadFile(Context context, HttpServletRequest request,
                 if (configurationService.getBooleanProperty("submission-curation.virus-scan"))
                 {
                     Curator curator = new Curator();
-                    curator.addTask("vscan").curate(item);
+                    curator.addTask("vscan").curate(context, item.getHandle());
```

Comment:
Doesn't work, eh?  Maybe the Handle has not been created yet.

curate(Context, String) just resolves the handle to a DSO again.  I think we need a new Curator.curate(Context, DSpaceObject) which sets the curation context (curationCtx) to that Context and then delegates to Curator.curate(DSpaceObject).  Then we don't have to resolve a handle that might not even exist yet, just to get back the DSO that we already have.  (There may also be concurrency issues with HandleService.resolveToObject(Context, DSpaceObject.getHandle()).)